compiler/src/dotty/tools/dotc/printing/Formatting.scala

```diff
@@ -52,10 +55,10 @@ object Formatting {
       inline def apply[A](using inline z: Show[A]): Show[A] = z
 
       given [X: Show]: Show[Seq[X]] with
-        def show(x: Seq[X]) = x.map(Show[X].show)
+        def show(x: Seq[X]) = (ctx ?=> x.map(show1)): CtxShow
```

Comment:
I found this code very puzzling. I figured out eventually that it's a shorthand for
```scala
new CtxShow:
  def run(using Context) = x.map(show1)
```
It's longer but makes it much clearer what goes on. Consider using the expansion instead (also for the next one)